packages/cheqd/src/ledger/CheqdLedgerService.ts

```diff
@@ -0,0 +1,123 @@
+
+import { injectable } from '@aries-framework/core'
+import { createCheqdSDK, DIDModule, ResourceModule, CheqdNetwork } from '@cheqd/sdk'
+
+import { CheqdModuleConfig } from '../CheqdModuleConfig'
+import { parseCheqdDid } from '../anoncreds/utils/identifiers'
+import { getCosmosPayerWallet } from '../dids/didCheqdUtil'
+
+export interface ICheqdLedgerConfig {
+  network: string
+  rpcUrl: string
+  readonly cosmosPayerWallet: Promise<DirectSecp256k1HdWallet | DirectSecp256k1Wallet>
+  sdk?: CheqdSDK
+}
+
+export enum DefaultRPCUrl {
+  Mainnet = 'https://rpc.cheqd.net',
+  Testnet = 'https://rpc.cheqd.network',
+}
+
+@injectable()
+export class CheqdLedgerService {
+  private networks: ICheqdLedgerConfig[]
+
+  public constructor(cheqdSdkModuleConfig: CheqdModuleConfig) {
+    this.networks = cheqdSdkModuleConfig.networks.map((config) => {
+      const { network, rpcUrl, cosmosPayerSeed } = config
+      return {
+        network,
+        rpcUrl: rpcUrl ? rpcUrl : network === CheqdNetwork.Mainnet ? DefaultRPCUrl.Mainnet : DefaultRPCUrl.Testnet,
+        cosmosPayerWallet: getCosmosPayerWallet(cosmosPayerSeed),
+      }
+    })
+  }
+
+  public async connect() {
+    for (const network of this.networks) {
+      network.sdk = await createCheqdSDK({
+        modules: [DIDModule as unknown as AbstractCheqdSDKModule, ResourceModule as unknown as AbstractCheqdSDKModule],
+        rpcUrl: network.rpcUrl,
+        wallet: await network.cosmosPayerWallet.catch(() => {
+          throw new Error(`[did-provider-cheqd]: valid cosmosPayerSeed is required`)
+        }),
+      })
+    }
+  }
+
+  private getSdk(did: string) {
+    const parsedDid = parseCheqdDid(did)
+    if (!parsedDid) {
+      throw new Error('Invalid DID')
+    }
+    if (this.networks.length === 0) {
+      throw new Error('No cheqd networks configured')
+    }
+
+    const network = this.networks.find((network) => network.network === parsedDid.network)
+    if (!network || !network.sdk) {
+      throw new Error('Network not configured')
+    }
+    return network.sdk
+  }
+
+  public async create(
+    didPayload: DIDDocument,
+    signInputs: SignInfo[],
+    versionId?: string | undefined,
+    fee?: DidStdFee
+  ) {
+    return await this.getSdk(didPayload.id).createDidDocTx(signInputs, didPayload, '', fee, undefined, versionId)
+  }
+
+  public async update(
+    didPayload: DIDDocument,
+    signInputs: SignInfo[],
+    versionId?: string | undefined,
+    fee?: DidStdFee
+  ) {
+    return await this.getSdk(didPayload.id).updateDidDocTx(signInputs, didPayload, '', fee, undefined, versionId)
+  }
+
+  public async deactivate(
+    didPayload: DIDDocument,
+    signInputs: SignInfo[],
+    versionId?: string | undefined,
+    fee?: DidStdFee
+  ) {
+    return await this.getSdk(didPayload.id).deactivateDidDocTx(signInputs, didPayload, '', fee, undefined, versionId)
+  }
+
+  public async resolve(did: string, version?: string) {
```

Comment:
Is It LedgerService part? Maybe let's create CheqdDidResolverService?

Reply:
Its querying from the ledger using the sdk, don't need a specfic service just for this.